File: src/main/java/it/plague/blog/serializer/LocalDateDeserializer.java
package it.plague.blog.serializer;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;

import java.io.IOException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class LocalDateDeserializer extends StdDeserializer<LocalDateTime> {

  private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

  public LocalDateDeserializer(Class<?> clazz) {
    super(clazz);
  }

  @Override
  public LocalDateTime deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
    var text = jsonParser.readValueAs(String.class);
    return text == null ? LocalDateTime.MAX : LocalDateTime.parse(text, formatter);
  }
}

File: src/main/kotlin/it/plague/blog/serializer/LocalDateDeserializer.kt
package it.plague.blog.serializer

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.StdDeserializer
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class LocalDateDeserializer : StdDeserializer<LocalDateTime>(LocalDateTime::class.java) {

	private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")

	override fun deserialize(jsonParser: JsonParser, context: DeserializationContext): LocalDateTime {
		val text = jsonParser.readValueAs(String::class.java)
		return if (text == null) LocalDateTime.MAX else LocalDateTime.parse(text, formatter)
	}
}
